### ISSUE TITLE
Release v1.99.1 missing API version 2022-11-01-preview

### ISSUE THREAD

**Issue by source-studio** [pulumi/pulumi-azure-native#2364]
### What happened?

The forthcoming Azure Container Apps preview release appears to be using API version `2022-11-01-preview`, which was present in release v1.99.0 but is now missing in v1.99.1 due to this recent change to reduce the NodeJS package size:

https://github.com/pulumi/pulumi-azure-native/blob/e8f28acd0871c42ab185996a765c4978ec4ba41c/versions/removed.json#L3

### Expected Behavior

The continued presence of this API version - otherwise, forthcoming preview features such as [managed certificates](https://github.com/microsoft/azure-container-apps/issues/607) will not be available.

### Steps to reproduce

1. Restore v1.99.0 of the `Pulumi.AzureNative` NuGet package.
2. Note that the `Pulumi.AzureNative.App.V20221101Preview` namespace exists.
3. Restore v1.99.1 of the NuGet package.
4. Note that the namespace no longer exists.

### Output of `pulumi about`

N/A

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by kpitzen** [comment]
Hi @source-studio - thank you for bringing this to our attention. While we are aware missing API versions can be a frustration, we're unfortunately constrained by NPM publishing limits with respect to file counts. This will be resolved in the next major version of this provider (2.0), in which we will re-add new versions of APIs as they are released.

I'll keep this open for now, but it is directly blocked by
https://docs.google.com/document/d/1dERljH0H5KXZ69ed6sSwwkPLQMLcK2IrWyCckGV5-6I/edit#
Thanks!

**Comment by source-studio** [comment]
Thanks for your response.

When do you expect v2 will be released? Will this perhaps introduce a more modular approach to the NuGet packages? The current one weighs in at 180Mb or so!

Just out of interest, why was this particular preview API version chosen for removal and not others?

Thanks.

**Comment by source-studio** [comment]
Thanks, that will be a real improvement if the size of the package can be reduced in v2 onwards.

However, I’m still confused by the choice to remove that particular API version in the case of the `Microsoft.App` namespace. `2022-01-01-preview` is the _oldest_ version, and this perhaps should have been the one to have been removed if pruning was required, rather than the _latest_ one (`2022-11-01-preview`).

My understanding is that Microsoft are about to announce a public preview of a several new features that fall under this API version, which means we will essentially be blocked from using Pulumi, because any updates will not be propagated into the NuGet package.

Is there any chance you could review this?

**Comment by kpitzen** [comment]
I don't believe we will be able to remove older versions of services at this time - that's a breaking change, and would force us to release a new major version of the provider. In this case, 1.99.0 should likely not have included that version in the first place, since now 1.99.0 -> 1.99.1 is breaking, which is not ideal.

**Comment by source-studio** [comment]
I can see your dilemma. However, I would argue that in this particular situation, those that are more likely to upgrade to later versions of the provider (like myself) are seeking to use the more recent preview releases and have therefore been broken by this change for at least 6 weeks until v2 is released. In the case of the `Microsoft.App` namespace, there is a clear upgrade path from the oldest API version if that were to be pruned. For the few that are still using this older version _and_ choose to upgrade to the latest version of the provider, this can be rectified. However, for those “living on the edge” at this point, there’s currently no path available, so it essentially represents a showstopper. The choice is therefore to frustrate a few people, but with the consolation that they have a path forward, or frustrate (perhaps more) people by enforcing a hard cut-off on preview API versions, albeit temporarily.

If the plan is to prune older API versions anyway, would you consider bringing that work forward in this case, and releasing v1.100? Like you say though, this would represent another breaking change.

Alternatively, since this situation was solely caused by NPM, why not apply the API version removals only in the case of the NPM package for now? At least consumers of the provider working in other languages are not unnecessarily affected by a limitation that is irrelevant to their ecosystem, and the blast radius is therefore reduced. This solution will probably ensure that the most people are happy, until v2 is released.

**Comment by kpitzen** [comment]
I don't think we can alter our plans to bring the work forward, though we do intend to ship prerelease versions (with no SLAs) much earlier than the final 2.0 release. An alternate workaround for the time being would be to hard-pin to 1.99.0, correct?

**Comment by kpitzen** [comment]
@source-studio Also for context - we plan to ship the first prerelease version, including all the older service removals and newer service inclusions some time in the next couple of weeks. It's the top priority for my team, so expect news very soon.  Thanks again!

**Comment by source-studio** [comment]
That’s correct, but there is a bug within the REST spec of this particular preview API version that we have reported and are hoping will be rectified soon, which is blocking our usage of the feature. But even if Microsoft do fix this, it will now not be propagated to Pulumi, at least not soon, so you see our dilemma.

Thanks for the clarification about v2 prereleases - I’ll look out for them!

**Comment by rhuanbarreto** [comment]
Managed to use this by having v1 and v2 provider coexisting in the same project by adding this in the package.json dependencies:

```
"@pulumi/azure-native-v2": "npm:@pulumi/azure-native@v2.0.0-beta.1",
```

**Comment by mikhailshilkov** [comment]
This issue is the same as #2444, which is now closed by 2.0 release. See https://github.com/pulumi/pulumi-azure-native/issues/2444#issuecomment-1633973959
